taps_to_cases/runner/src/main/java/com/microsoft/hydralab/t2c/runner/finder/AndroidElementFinder.java

```diff
@@ -0,0 +1,49 @@
+package com.microsoft.hydralab.t2c.runner.finder;
+
+import com.google.common.base.Strings;
+import com.microsoft.hydralab.t2c.runner.controller.BaseDriverController;
+import com.microsoft.hydralab.t2c.runner.elements.AndroidElementInfo;
+import org.openqa.selenium.WebElement;
+
+public class AndroidElementFinder implements ElementFinder<AndroidElementInfo> {
+
+    private final BaseDriverController driverController;
+
+    public AndroidElementFinder(BaseDriverController driverController) {
+        this.driverController = driverController;
+    }
+
+    @Override
+    public WebElement findElement(AndroidElementInfo elementInfo) {
+        WebElement elementFound;
+        if (!Strings.isNullOrEmpty(elementInfo.getContentDesc())) {
+            elementFound = driverController.findElementByAccessibilityId(elementInfo.getContentDesc());
+            if (elementFound != null) {
+                return elementFound;
+            }
+        }
+
+
+        if (!Strings.isNullOrEmpty(elementInfo.getResourceId())) {
```

Comment:
It will be good if you can help to move the find by id up as the first attribute to check since the it will be faster if there is an available id